src/System.IO.Pipelines/StreamExtensions.cs

```diff
@@ -78,9 +64,18 @@ public static IPipeReader AsPipelineReader(this Stream stream, CancellationToken
         /// <param name="stream"></param>
         /// <param name="writer"></param>
         /// <returns></returns>
-        public static Task CopyToAsync(this Stream stream, IPipeWriter writer)
+        public static async Task CopyToAsync(this Stream stream, IPipeWriter writer)
         {
-            return stream.CopyToAsync(new PipelineWriterStream(writer));
+            try
+            {
+                await stream.CopyToAsync(new PipelineWriterStream(writer));
+            }
+            catch (Exception ex)
+            {
+                writer.Complete(ex);
+                return;
+            }
+            writer.Complete();
```

Comment:
My problem then is the Execute is more you want it to finish the pipe because you are actively wrapping the stream but for the copy to you don't want to. In both cases you want to complete the pipeline if there is an exception. I think the best idea is to just return the task from the copytoasync and catch and close in the execute.